test/unit/objects/p5.Color.js

```diff
@@ -553,6 +555,34 @@ suite('p5.Color', function() {
     });
   });
 
+  suite('new p5.Color in HSB-360 mode', function() {
+    setup(function() {
+      myp5.colorMode(myp5.HSB, 360, 100, 100, 1);
+      c = myp5.color(48, 100, 100, 1);
+    });
+
+    test('should correctly set HSBA value', function(){
+      assert.deepEqual(c.hsba, [48, 100, 100, 1]);
+    });
+
+    test('should correctly set converted value', function(){
+      assert.deepEqual(c._converted_color, [34, 255, 255, 255]);
+    });
+
+    test('should return passed, unconverted, hue value', function(){
+      assert.equal(c.getHue(), 48);
+    });
+
+    test('should return passed, unconverted, saturation value', function(){
+      assert.equal(c.getSaturation(), 100);
+    });
+
+    test('should return passed, unconverted, brightness value', function(){
+      assert.equal(c.getBrightness(), 100);
+    });
+
+  });
+
```

Comment:
Might be worth adding another suite to test an arbitrary colorMode range, like `HSB, 1000, 360, 720, 10` or something, since we provide the ability to set non-standard ranges.